zetaclient/chains/solana/observer/inbound.go

```diff
@@ -167,30 +168,32 @@ func FilterInboundEvents(
 	seenCall := false
 	events := make([]*clienttypes.InboundEvent, 0)
 
-	// loop through instruction list to filter the 1st valid event
-	for i, instruction := range tx.Message.Instructions {
+	// helper to parse any instruction
+	parseInstruction := func(instruction solana.CompiledInstruction, location string) error {
```

Comment:
Do you think it makes sense to return a list of events here and move this out to a spearate function ? It would improve the testability

Reply:
Looks like a good point, the function now spans 170+ lines. If we could create a separate func
`func parseInstructions(instructions []solana.CompiledInstruction) ([]*clienttypes.InboundEvent, error)`.

The whole input will be: `tx.Message.Instructions + inner.Instructions` and the flags `seenDeposit, seenDepositSPL and seenCall` can live inside `parseInstructions`. Not sure how this will impact your unit tests?